pkg/mcp/kubernetes.go

```diff
@@ -262,20 +259,40 @@ func (k *kubernetesBackend) k8sObjectsForUVXOrNPX(id string, server ServerConfig
 		}
 	}
 
-	var err error
-	nanobotFileStringData["nanobot.yaml"], err = constructNanobotYAML(serverDisplayName, server.Command, server.Args, secretStringData)
-	if err != nil {
-		return nil, fmt.Errorf("failed to construct nanobot.yaml: %w", err)
+	if len(server.Args) > 0 {
+		// Copy the args to avoid modifying the original slice.
+		args := make([]string, len(server.Args))
+		for i, arg := range server.Args {
+			args[i] = expandEnvVars(arg, fileMapping, nil)
+		}
+
+		server.Args = args
 	}
 
-	objs = append(objs, &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:        name.SafeConcatName(id, "run"),
-			Namespace:   k.mcpNamespace,
-			Annotations: annotations,
-		},
-		StringData: nanobotFileStringData,
-	})
+	if server.Runtime == types.RuntimeContainerized {
+		if server.Command != "" {
+			command = []string{expandEnvVars(server.Command, fileMapping, nil)}
+		}
+
+		image = expandEnvVars(server.ContainerImage, fileMapping, nil)
+		args = server.Args
```

Comment:
We also need to set `port` to `server.ContainerPort`.

Reply:
Nice catch! I was just using 8099 in my testing, so didn't notice this. I updated and did another test with a different port to ensure it worked.